Sources/AuthFoundation/Utilities/JSONValue.swift

```diff
@@ -251,13 +251,13 @@ extension JSON: Codable {
 
 fileprivate extension NSNumber {
     var isFloatingPoint: Bool {
-        let type = CFNumberGetType(self as CFNumber)
-        switch type {
-        case .floatType, .float32Type, .float64Type, .cgFloatType, .doubleType:
+        if strcmp(objCType, "f") == 0 ||
+            strcmp(objCType, "d") == 0
+        {
```

Comment:
interesting, how sure are you that this will work permanently?

Reply:
https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/ObjCRuntimeGuide/Articles/ocrtTypeEncodings.html

This has been around within Objective-C since the dawn of time, so it's unlikely to change.  I would use the `@encode` feature, but that's not available in Swift, so it was more straightforward to use the string equivalents.